src/fireedge/src/client/bootstrap.js

```diff
@@ -0,0 +1,316 @@
+
+  const useLocalFallback = !!localStorage.getItem(FORCE_LOCAL_FALLBACK)
+
+  const syncLocalCopy = () => {
+    localStorage.setItem(
+      LOCAL_REMOTES_COPY,
+      JSON.stringify(remotesConfig, null, 2)
+    )
+  }
+
+  const localCopyConfig = JSON.parse(localStorage.getItem(LOCAL_REMOTES_COPY))
+
+  if (!isInitialized(localCopyConfig) || (!fallback && !useLocalFallback)) {
+    syncLocalCopy()
+  }
+
+  if (!forceNoDialog && (fallback || useLocalFallback)) {
+    await editFallbackConfig()
+  }
+
+  fallback
+    ? localStorage.setItem(USING_FALLBACK, true)
+    : localStorage.removeItem(USING_FALLBACK)
+}
+
+const testLoadModule = async (scriptUrl) =>
+  new Promise((resolve, reject) => {
+    const script = document.createElement('script')
+
+    script.type = 'text/javascript'
+
+    script.src = scriptUrl
+    script.async = true
+
+    script.onload = () => {
+      resolve()
+    }
+
+    script.onerror = () => {
+      reject(new Error(`Failed to load script: ${scriptUrl}`))
+    }
+
+    window.addEventListener(
+      'error',
+      (event) => {
+        reject(
+          new Error(
+            `Runtime error in script: ${event?.filename || scriptUrl} - ${
+              event.message
+            }`
+          )
+        )
+      },
+      { once: true }
+    )
+
+    document.head.appendChild(script)
+  })
+
+const checkRemotes = async () => {
+  const failedModules = []
+  const remotesConfig = JSON.parse(localStorage.getItem(LOCAL_REMOTES_COPY))
+
+  const loadPromises = Object.entries(remotesConfig).map(
+    async ([id, module]) => {
+      const isValid = module?.name && module?.entry
+      if (!isValid) {
+        throw new Error(
+          `Module: ${id} contains invalid configuration. Ensure 'name' and 'entry' are defined: ${JSON.stringify(
+            module,
+            null,
+            2
+          )}`
+        )
+      }
+
+      try {
+        await testLoadModule(module.entry)
+      } catch (error) {
+        failedModules.push(id)
+        throw new Error(`Script loading error in ${id}: ${error}`)
+      }
+    }
+  )
+
+  await Promise.all(loadPromises)
+
+  if (failedModules?.length > 0) {
+    throw new Error(
+      `The following modules failed to load: ${failedModules.join(', ')}`
+    )
+  }
+
+  return true
+}
+
+const loadClient = async () => {
+  // eslint-disable-next-line no-undef
+  await __webpack_init_sharing__('default')
+
+  const { default: initApp } = await import('client/sunstone')
+  initApp()
+}
+
+const syncConfig = async () => {
+  window.__REMOTES_MODULE_CONFIG__ = await JSON.parse(
+    localStorage.getItem(LOCAL_REMOTES_COPY)
+  )
+}
+
+async function bootstrap(forceNoDialog = false) {
+  try {
+    await initLocalRemotesConfig(forceNoDialog)
+    await checkRemotes()
+    await syncConfig()
+    loadClient()
+  } catch (error) {
+    const failedModule = error?.message.match(/modules\/([^/]+)/)?.[1] || ''
+    console.error('Failed to load module: ', failedModule || error)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Log Injection occurs when untrusted user input is directly written to log files without proper sanitization. This can allow attackers to manipulate log entries, potentially leading to security issues like log forging or cross-site scripting. To prevent this, always sanitize user input using `encodeURIComponent()` or `DOMPurify.sanitize()` before logging.  Learn more - https://cwe.mitre.org/data/definitions/117.html